src/Aspire.Hosting/Orchestrator/ParameterProcessor.cs

```diff
@@ -352,5 +362,71 @@ await notificationService.PublishUpdateAsync(parameter, s =>
                 }
             }
         }
+
+        // Save deployment state at the end if modified
+        if (stateModified && deploymentState is not null)
```

Comment:
Are we properly saving this state in publish mode? If parameters change (new ones added, old ones removed etc), there should be an update to deployment state.